### PR TITLE
server/gc_worker: Replace Futuer::Worker with Poll::Worker

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

### What problem does this PR solve?

Issue Number: Part of https://github.com/tikv/tikv/issues/10641

### What is changed and how it works?

Replace `Futuer::Worker` with `Poll::Worker`

### Check List 

Tests 

- Unit test
- Integration test

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by Xuanwo** [comment]
Unittest for `test_compaction_filter_handle_deleting` is broken, please give me some advice for the fix.

File: src/server/gc_worker/compaction_filter.rs [hunk 1]
```diff
@@ -690,21 +691,24 @@ pub mod test_utils {
...
+
             TestGCRunner {
                 safe_point,
                 ratio_threshold: None,
                 start: None,
                 end: None,
                 target_level: None,
-                gc_scheduler: FutureScheduler::new("test-gc-sched", tx),
+                gc_scheduler,
                 gc_receiver: rx,
```
Comment (5kbpers) [inline src/server/gc_worker/compaction_filter.rs:712]: Should be `worker.receiver`?

Comment (Xuanwo) [reply src/server/gc_worker/compaction_filter.rs:712]: Do you mean implement `pub fn receiver(&self) -> Option<UnboundedReceiver<Msg<T>>>` for `LazyWorker`?

**Comment by Xuanwo** [comment]
There are three failures for now:

```
[2021-08-01T04:02:29.551Z] failures:
[2021-08-01T04:02:29.551Z]     cases::test_gc_worker::test_error_in_compaction_filter
[2021-08-01T04:02:29.551Z]     cases::test_merge::test_node_merge_cascade_merge_with_apply_yield

[2021-08-01T04:02:19.464Z] failures:
[2021-08-01T04:02:19.464Z]     server::gc_worker::compaction_filter::tests::test_compaction_filter_handle_deleting
```

I'm working on the fix.

**Comment by Xuanwo** [comment]
ping @5kbpers, all unit tests have been passed, PTAL.

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
/cc @5kbpers 

@Xuanwo I see your [post](https://internals.tidb.io/t/topic/307) on the forum and it seems @5kbpers already the reviewer on this thread. You can `/cc @reviewer` to ping him so that one can filter his own pending review on https://github.com/pulls/review-requested etc.

**Review by 5kbpers — APPROVED** [review]
Sorry for so late replying. Thank you!

**Pull request merged**